src/ensembl/src/shared/tooltip/Tooltip.tsx

```diff
@@ -81,50 +83,55 @@ const Tooltip = (props: Props) => {
   }, []);
 
   useEffect(() => {
-    const node = tooltipElementRef.current;
-    const parentElement = node && node.parentElement;
-    if (!(node && parentElement)) {
+    if (isWaiting) {
       return;
     }
-    parentRef.current = parentElement;
 
-    setInlineStyles(getInlineStyles({ ...props, parentElement }));
+    const tooltipElement = tooltipElementRef.current;
+    const parentElement = tooltipElement && tooltipElement.parentElement;
 
-    if (!props.autoAdjust) {
+    if (!(tooltipElement && parentElement)) {
```

Comment:
Do we need `tooltipElement` here?

Reply:
We are passing `tooltipElement`, along with `parentElement` to the `adjustPosition` function below. `adjustPosition` relies on the `tooltipElement` being of type `HTMLDivElement`, but here, since `tooltipElement` is `tooltipElementRef.current`, it can be `HTMLDivElement` or `null`. So I would think yes, we want to assure Typescript that both `tooltipElement` and `parentElement` are not null before we proceed further.

Reply:
Cool, I can see that Typescript is not happy if we remove it. I was thinking that if `tooltipElement` is null, `parentElement` will also be null in which case the `if` condition will return true even if we just check for `!parentElement`.